### ISSUE TITLE
Error after300 sec

### ISSUE THREAD

**Issue by Benoulimou** [flashnuke/wifi-deauth#23]
Hi, Could you tell me what this error means? It happens all the time after about 300 seconds.

`[!] Exception in deauth-loop -> Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/wifi_deauth/wifi_deauth.py", line 321, in _run_deauther
    self._send_deauth_client(ap_mac, client_mac)
  File "/usr/local/lib/python3.11/dist-packages/wifi_deauth/wifi_deauth.py", line 335, in _send_deauth_client
    sendp(RadioTap() /
  File "/usr/lib/python3/dist-packages/scapy/sendrecv.py", line 480, in sendp
    return _send(
           ^^^^^^
  File "/usr/lib/python3/dist-packages/scapy/sendrecv.py", line 417, in _send
    results = __gen_send(socket, x, inter=inter, loop=loop,`

thx

**Comment by flashnuke** [comment]
Hi
Does the de-auth work during those 300 seconds?

Can you please share the full exception log? (It seems to be truncated)

**Comment by Benoulimou** [comment]
yes, it works Can you tell me where the log is? I don't see anything in the file

**Comment by flashnuke** [comment]
I mean from the output, as the snippet you provided doesn't seem to be full (it gets truncated at `results = __gen_send(socket, x, inter=inter, loop=loop,`)

Also, which OS are you using? Is it Nethunter (Kali on Android) by any chance?

**Comment by Benoulimou** [comment]
![image](https://github.com/user-attachments/assets/9a1a03b6-f527-45e2-8887-94a50e743341)
that's all I have, I'm on kali linux on vm

**Comment by Benoulimou** [comment]
![image](https://github.com/user-attachments/assets/364d0d72-21e8-42e3-b036-7b496f053f83) 
here's another one

**Comment by Benoulimou** [comment]
![image](https://github.com/user-attachments/assets/cd8328c1-3c2d-4ada-991c-c366e279fcca)
another one after 1096 sec on root user

**Comment by flashnuke** [comment]
Seems like an unhandled exception that comes from  scapy library... 
I will add a try-except mechanism that should help overcome temporary exceptions (I assume this is temporary as the next time you run the script it runs ok)

I will deploy this in the next version, meanwhile I'd suggest to try connecting the network interface to a different USB port

**Comment by Benoulimou** [comment]
ok, 
J'ai voulu l'installé sur une autre machine kali linux et j'ai cette erreur que je n'arrive pas a bypasser :

`pip3 install
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Kali-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have pypy3-venv installed.

    If you wish to install a non-Kali-packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

    For more information, refer to the following:
    * https://www.kali.org/blog/python-externally-managed/
    * /usr/share/doc/python3.12/README.venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.


`

**Comment by flashnuke** [comment]
try setting up venv:

```bash
python3 -m venv venv
source venv/bin/activate
pip3 install .
```

**Comment by Benoulimou** [comment]
![image](https://github.com/user-attachments/assets/343b7204-1086-4624-9139-9d2f362998bf)

Sorry don't work

**Comment by flashnuke** [comment]
You need to create the venv inside the wifi-deauth directory

```bash
git clone https://github.com/flashnuke/wifi-deauth.git
cd wifi-deauth
// Create venv and install here (Using the commands I provided above)
```

**Comment by flashnuke** [comment]
> great it's work ! thk !
> 
> But then I have to stay on venv for it to work? 

Your welcome!


Yes, you should stay on venv 
This limitation is imposed by the new Kali machine it seems

**Comment by flashnuke** [comment]
hey @Benoulimou 
I believe what you had is a network buffer overflow (due to network congestion - too many packets at a short period)
I addressed this in https://github.com/flashnuke/wifi-deauth/pull/24:

now, If a deauth attempt fails - the program would throttle down between each failed attempt, this would allow the buffer to deplete (if the successive attempts fail also - the program will eventually exit as it did before)
also I fixed the malformed exception prints

**Comment by Benoulimou** [comment]
Great! 
this is the error I get on the 1st launch, then when I relaunch it works.
![image](https://github.com/user-attachments/assets/e0cd59f9-fca5-4203-8c8d-32acb1ca2ff8)

**Comment by flashnuke** [comment]
Just to clarify, does this happen every time (i.e when you reconnect the wifi dongle) or just once?

**Comment by Benoulimou** [comment]
no, I run it, I get the error, I run it again, I don't get the error and I get it every time

**Comment by flashnuke** [comment]
can you please pull latest, run with debug mode and share the output? i want to see if your interface it set up correctly
```bash
git pull
sudo python3 wifi_deauth.py -i wlan0 -d  # use the '-d' flag for debug mode
```

**Comment by Benoulimou** [comment]
I'm back, 
I've installed debian 10, I don't want to use env but I still have an error with pip3 which doesn't work and python3 can't find install Can you help me?

**Comment by DDJ-MooN** [comment]
Bonjour
À propos d'une autre problème :
Voici une explication détaillée de l'erreur :

# Cause de l'erreur
L'erreur se produit après environ 300 secondes d'exécution du programme
wifi_deauth, qui est utilisé pour tester la sécurité des réseaux sans fil.
L'erreur est générée par une exception dans la boucle de déconnexion
(_deauth-loop).

# Détails techniques
1. Fichier affecté : (link unavailable) (ligne 321).
2. Méthode affectée : _run_deauther.
3. Erreur : Échec de l'envoi d'un paquet RadioTap via Scapy.

# Causes possibles
1. Problèmes de connexion réseau.
2. Problèmes de compatibilité entre les versions de Scapy et wifi_deauth.
3. Problèmes de sécurité ou de pare-feu.
4. Erreurs de configuration.
5. Problèmes de droits d'accès.

# Solutions possibles
1. Mettre à jour Scapy : `pip install --upgrade scapy`.
2. Mettre à jour wifi_deauth : `pip install --upgrade wifi_deauth`.
3. Vérifier la stabilité de la connexion réseau.
4. Vérifier les paramètres de configuration.
5. Exécuter le programme avec des droits d'administrateur.
6. Redémarrer le programme.
7. Vérifier les mises à jour du système.

# Ressources
1. Documentation wifi_deauth.
2. Documentation Scapy.
3. Forum Stack Overflow.
4. GitHub.

    Merci a votre attention
